driver/smooth_l1loss_driver.hpp

```diff
@@ -0,0 +1,366 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2024 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#ifndef GUARD_MIOPEN_SMOOTH_L1LOSS_DRIVER_HPP
+#define GUARD_MIOPEN_SMOOTH_L1LOSS_DRIVER_HPP
+
+#include "InputFlags.hpp"
+#include "driver.hpp"
+#include "miopen/errors.hpp"
+#include "tensor_driver.hpp"
+#include "timer.hpp"
+
+#include <../test/ford.hpp>
+#include <../test/tensor_holder.hpp>
+#include <../test/verify.hpp>
+
+#include <miopen/miopen.h>
+#include <miopen/tensor.hpp>
+
+#include <vector>
+
+#ifndef MLO_SMOOTH_L1LOSSMHOST_H_
+#define MLO_SMOOTH_L1LOSSMHOST_H_
+
+template <typename Tgpu, typename Tcheck>
+int32_t mloSmoothL1LossForwardRunHost(const miopenTensorDescriptor_t tensorDesc,
+                                      const Tgpu* input,
+                                      const Tgpu* target,
+                                      Tcheck* outputhost,
+                                      const miopenLossReduction_t reduction,
+                                      const float beta)
+{
+    size_t size = miopen::deref(tensorDesc).GetElementSize();
+
+    auto loss_no_reduce = [&]() {
+        par_ford(size)([&](size_t i) {
+            auto diff     = static_cast<Tcheck>(abs(input[i] - target[i]));
+            outputhost[i] = diff < beta ? 0.5f * diff * diff / beta : diff - 0.5f * beta;
+        });
+    };
+
+    auto loss_mean_reduce = [&]() { std::cout << "Unsupported Mean Reduction"; };
+
+    auto loss_sum_reduce = [&]() { std::cout << "Unsupported Sum Reduction"; };
+
+    switch(reduction)
+    {
+    case MIOPEN_LOSS_MEAN_REDUCTION: loss_mean_reduce(); break;
```

Comment:
This looks excessive. The reader has to check what loss_mean_reduce do, to find it's just a print.

Reply:
Should I seperate the SmoothL1Loss API for each reduction types?

Reply:
That is not what I meant. I was talking about loss_mean_reduce. See line 64.
this line could be just like this;
`case MIOPEN_LOSS_MEAN_REDUCTION: std::cout << "Unsupported Mean Reduction"; break;`
And it doesn't do loss calculation, unlike its name, right?

Reply:
I have shorten it. I also did this to the cpu_smooth_l1loss.hpp too.